imperial.js

```diff
@@ -1,20 +1,62 @@
-import { Nation } from "./constants.js";
+import { Nation, Strings, NationCard } from "./constants.js";
 import Action from "./action.js";
 
+Map.fromIterable = (iter, getValue) => {
+  const m = new Map();
+  for (const k of iter) {
+    m.set(k, getValue(k));
+  }
+  return m;
+};
+
+Array.fromIterable = (iter, getValue) => {
+  const a = [];
+  for (const k of iter) {
+    a.push(getValue(k));
+  }
+  return a;
+};
+
 export default class Imperial {
-  static fromLog(log) {
-    let game = new Imperial();
+  static fromLog(log, initialState) {
```

Comment:
How do you see `initialState` being most useful as a thing to pass in here? What does this enable for us?